dartagnan/src/main/java/com/dat3m/dartagnan/wmm/utils/Flag.java

```diff
@@ -14,8 +12,8 @@ public enum  Flag {
     // TODO: Add linux when implemented
     public static ImmutableSet<Flag> all = ImmutableSet.of(ARM_UNPREDICTABLE_BEHAVIOUR);
 
-    public BooleanFormula repr(SolverContext ctx){
-    	return ctx.getFormulaManager().makeVariable(BooleanType, code());
+    public BooleanFormula repr(FormulaManager m){
```

Comment:
Why the change to `FormulaManager` here?
Is it because you create some new variable that is not directly supported by the `EncodingContext`?
I would still provide the encoding context though.

Reply:
This is a remainder of a change I missed to clean up.  This method would never need to create a prover environment, or fetch statistics of the context.  Neither would it need to react to elements of the verification task.

Reply:
I do not plan on reverting this.  Actually, with this argument even `EncodingContext` would not need a reference to `SolverContext`.  I yet decided against it, because it is unnecessary.

Reply:
I'm not sure I can totally follow here. It is true that the full `EncodingContext` is not needed here, but that can also be said about the `FormulaManager`: the code just requires a `BooleanFormulaManager` to generate its flag variable.
So then why choose this "in-between" solution?

Reply:
I used automatic refactoring.  Although it is not used in this particular place, `BooleanFormulaManager` does not provide methods like `escape(String)` which would be needed on other occasions.